### PR TITLE
Java: Refactor more queries to the new DataFlow module API

### PR DESCRIPTION
**Pull request by egregius313**
Refactors 17 more queries

### CODE REVIEW

File: java/ql/lib/semmle/code/java/security/XxeLocalQuery.qll [hunk 1]
```diff
@@ -21,3 +23,17 @@ class XxeLocalConfig extends TaintTracking::Configuration {
     any(XxeAdditionalTaintStep s).step(n1, n2)
   }
 }
+
+module XxeLocalConfig implements DataFlow::ConfigSig {
```
Comment (aschackmull) [inline java/ql/lib/semmle/code/java/security/XxeLocalQuery.qll:27]: Yes, this should have qldoc.

File: java/ql/lib/semmle/code/java/security/XxeLocalQuery.qll [hunk 2]
```diff
@@ -21,3 +23,17 @@ class XxeLocalConfig extends TaintTracking::Configuration {
...
+  predicate isSource(DataFlow::Node src) { src instanceof LocalUserInput }
```
[the remark this answers is not in the document]
Comment (aschackmull) [reply java/ql/lib/semmle/code/java/security/XxeLocalQuery.qll:28]: (I.e. for these you can disregard the CI warning).

File: java/ql/lib/semmle/code/java/security/XxeLocalQuery.qll [hunk 3]
```diff
@@ -21,3 +23,17 @@ class XxeLocalConfig extends TaintTracking::Configuration {
...
+  predicate isSink(DataFlow::Node sink) { sink instanceof XxeSink }
+
+  predicate isBarrier(DataFlow::Node sanitizer) { sanitizer instanceof XxeSanitizer }
+
+  predicate isAdditionalFlowStep(DataFlow::Node n1, DataFlow::Node n2) {
+    any(XxeAdditionalTaintStep s).step(n1, n2)
+  }
+}
+
+module XxeLocalFlow = TaintTracking::Make<XxeLocalConfig>;
```
Comment (aschackmull) [inline java/ql/lib/semmle/code/java/security/XxeLocalQuery.qll:39]: This is not caught by the CI check, but probably also deserves qldoc.

**Comment by aschackmull** [comment]
A couple of minor nitpicks, otherwise LGTM.

File: java/ql/src/Security/CWE/CWE-297/UnsafeHostnameVerification.ql [hunk 4]
```diff
@@ -29,7 +28,7 @@ private predicate alwaysReturnsTrue(HostnameVerifierVerify m) {
 }
 
 /**
- * A class that overrides the `javax.net.ssl.HostnameVerifier.verify` method and **always** returns `true` (though it could also exit due to an uncaught exception), thus
+ * A class that s the `javax.net.ssl.HostnameVerifier.verify` method and **always** returns `true` (though it could also exit due to an uncaught exception), thus
```
[the remark this answers is not in the document]
Comment (egregius313) [reply java/ql/src/Security/CWE/CWE-297/UnsafeHostnameVerification.ql]: Fixed.

**Pull request merged**